### PR TITLE
fix: Escape template literals and single quotes in codegen output

### PR DESCRIPTION
**Pull request by e-fisher**
## Description

Header values, cookie values, and URLs from recorded HAR traffic are embedded in JavaScript template literals when generating k6 scripts, but only request body content was being escaped. A malicious server could send response headers or cookies containing backticks or `\${}` syntax during recording, which would inject arbitrary JavaScript into the generated script.

This replaces `escapeBackticks` with `escapeTemplateLiteral`, which also escapes `\${}` interpolation sequences while preserving known safe patterns from the rules system (correlation_vars, VARS, file accessors, parameterization functions). Applied to URLs, headers, cookies, and request body content.

Check descriptions from verification rules are now escaped for single quotes to prevent syntax errors in generated code.

## How to Test

1. Record a session with a site that returns varied response headers
2. Generate a k6 script and verify the output is valid JavaScript
3. Add verification rules with special characters in string values, generate, and verify

## Checklist

- [x] I have performed a self-review of my code.
- [x] I have added tests for my changes.
- [ ] I have commented on my code, particularly in hard-to-understand areas.

## Related PR(s)/Issue(s)

Part of security hardening for critical findings from a codebase audit.

---

> [!NOTE]
> **Medium Risk**
> Security-sensitive codegen path with allowlist-based `${}` preservation; incorrect allowlist or escaping could break scripts or leave residual injection risk.
> 
> **Overview**
> Hardens **k6 script codegen** so recorded traffic cannot break out of generated JavaScript. **`escapeBackticks`** is replaced by **`escapeTemplateLiteral`**, which escapes backslashes, backticks, and **`${...}`** while leaving whitelisted rule-driven interpolations (e.g. **`correlation_vars`**, **`VARS`**, parameterization helpers) intact.
> 
> That escaping is now applied wherever recorded strings land in template literals: **URLs**, **headers**, **cookies**, and **non-binary request bodies** (not only bodies, which previously got partial escaping). **`escapeSingleQuotedString`** is added for **verification check** descriptions embedded in single-quoted keys.
> 
> New unit tests cover escape edge cases (bypass attempts, mixed safe/unsafe interpolations) and generated **`params`** / check output.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 6df5db1ffa4c560e9eb1d8b7e431790fb7caade9. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: src/codegen/codegen.ts [hunk 1]
```diff
@@ -354,8 +356,17 @@ export function isBinaryContent(content: string): boolean {
   return false
 }
 
-function escapeBackticks(content: string): string {
-  return content.replace(/`/g, '\\`')
+export function escapeTemplateLiteral(content: string): string {
+  return content
+    .replace(/`/g, '\\`')
+    .replace(
+      /\$\{(?!correlation_vars\[|VARS\[|getUniqueItem\(FILES\[|getParameterizationValue)/g,
```
[the remark this answers is not in the document]
Comment (e-fisher) [reply src/codegen/codegen.ts]: Real bug. The negative lookahead only checked the prefix, so ${VARS['x']+(()=>{ malicious })()} passed through unchanged. Replaced with sentinel-based approach: extract complete safe patterns (anchored to their closing syntax), escape all remaining ${, then restore. Added tests for the exact PoC from this comment.

File: src/codegen/codegen.ts [hunk 2]
```diff
@@ -354,8 +356,24 @@ export function isBinaryContent(content: string): boolean {
...
+const SAFE_INTERPOLATION =
+  /\$\{(?:correlation_vars\['[^']*'\]|VARS\['[^']*'\]|getUniqueItem\(FILES\['[^']*'\]\)\['[^']*'\]|getParameterizationValue\d+\(\))\}/g
+
+export function escapeTemplateLiteral(content: string): string {
+  const sentinels: string[] = []
+  const withSentinels = content.replace(SAFE_INTERPOLATION, (match) => {
+    sentinels.push(match)
+    return `__SAFE_INTERPOLATION_${sentinels.length - 1}__`
+  })
+  const escaped = withSentinels.replace(/`/g, '\\`').replace(/\$\{/g, '\\${')
```
[the remark this answers is not in the document]
Comment (e-fisher) [reply src/codegen/codegen.ts]: Real bug. Backslash before ${ produced \\${ which JS parses as literal backslash + live interpolation. Now escaping backslashes first (before backticks and ${). Added regression test for the PoC payload.

**Review by allansson — APPROVED** [review]
LGTM 🙌
